File: go/structural/Bridge/SodaImpSingleton.go
package bridge


//SodaImpSingleton - a Singleton to hold the current SodaImp

public class SodaImpSingleton {  
    private static SodaImp sodaImp
   
    public SodaImpSingleton(SodaImp sodaImpIn) {
        this.sodaImp = sodaImpIn
    }
    
    public static SodaImp getTheSodaImp() {
        return sodaImp
    }
}

File: scala/src/main/scala/Structural/Bridge/SodaImpSingleton.scala
package Structural.Bridge;


//SodaImpSingleton - a Singleton to hold the current SodaImp

class SodaImpSingleton {  
    private static SodaImp sodaImp;
   
    SodaImpSingleton(SodaImp sodaImpIn) {
        this.sodaImp = sodaImpIn;
    }
    
    static SodaImp getTheSodaImp() {
        return sodaImp;
    }
}